.github/workflows/workflow.yaml

```diff
@@ -0,0 +1,69 @@
+on:
+  workflow_call:
+    inputs:
+      build_type:
+        required: true
+        type: string
+
+jobs:
+  cpp-build:
+    name: C++ build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-cpp-matrix-build.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
+  cpp-tests-pr:
+    name: C++ tests for PRs
+    if: inputs.build_type == 'pull-request'
+    needs: cpp-build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-pr-cpp-tests.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
+  cpp-tests-nightly:
+    name: C++ tests for nightly builds
+    if: inputs.build_type == 'nightly'
+    needs: cpp-build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-nightly-cpp-tests.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
+  python-build:
+    name: Python build
+    needs: cpp-build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-python-matrix-build.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
+  python-tests-prs:
+    name: Python tests for PRs
+    if: inputs.build_type == 'pull-request'
+    needs: python-build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-pr-python-tests.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
+  python-tests-nightly:
+    name: Python tests for nightly builds
+    if: inputs.build_type == 'nightly'
+    needs: python-build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-nightly-python-tests.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
+  upload-conda-nightly:
+    name: Upload conda packages for nightly builds
+    if: inputs.build_type == 'nightly'
+    needs: [cpp-tests-nightly, python-tests-nightly]
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-upload-packages.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
+  upload-conda-branch:
+    name: Upload conda packages for branch builds
+    if: inputs.build_type == 'branch'
+    needs: python-build
+    secrets: inherit
+    uses: rapidsai/shared-action-workflows/.github/workflows/conda-upload-packages.yaml@feat/conda-actions
+    with:
+      build_type: ${{ inputs.build_type }}
```

Comment:
I wonder... do we need separate `upload` stages for `branch` builds vs. `nightly` builds? We should try experimenting with a single `upload` stage that uses `needs: [cpp-tests-nightly, python-tests-nightly]`. even though we skip tests on `branch` builds, I think the order of stages should still work correctly.

This would let us eliminate some duplication here.

Reply:
> even though we skip tests on branch builds, I think the order of stages should still work correctly.

I tested this theory and it didn't work. If we skip a job in the workflow, then any dependent jobs also get skipped.